syntax = "proto3";
package file;

service FileService {
  rpc Upload(UploadRequest) returns (UploadResponse);
  rpc Download(DownloadRequest) returns (DownloadResponse);
  rpc Delete(DeleteRequest) returns (DeleteResponse);
}

message UploadRequest {
  string name = 1;
  bytes data = 2;
}

message UploadResponse {
  string hash = 1;
}

message DownloadRequest {
  string hash = 1;
}

message DownloadResponse {
  bytes data = 1;
}

message DeleteRequest {
  string hash = 1;
}

message DeleteResponse {}
